tree/dataframe/inc/ROOT/RDF/RFilterWithMissingValues.hxx

```diff
@@ -0,0 +1,246 @@
+   }
+
+   bool CheckFilters(unsigned int slot, Long64_t entry) final
+   {
+      if (entry != fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()]) {
+         if (!fPrevNodePtr->CheckFilters(slot, entry)) {
+            // a filter upstream returned false, cache the result
+            fLastResult[slot * RDFInternal::CacheLineStep<int>()] = false;
+         } else {
+            // evaluate this filter, cache the result
+            const bool valueIsMissing = fValues[slot]->template TryGet<void>(entry) == nullptr;
+            if (fDiscardEntryWithMissingValue) {
+               valueIsMissing ? ++fRejected[slot * RDFInternal::CacheLineStep<ULong64_t>()]
+                              : ++fAccepted[slot * RDFInternal::CacheLineStep<ULong64_t>()];
+               fLastResult[slot * RDFInternal::CacheLineStep<int>()] = !valueIsMissing;
+            } else {
+               valueIsMissing ? ++fAccepted[slot * RDFInternal::CacheLineStep<ULong64_t>()]
+                              : ++fRejected[slot * RDFInternal::CacheLineStep<ULong64_t>()];
+               fLastResult[slot * RDFInternal::CacheLineStep<int>()] = valueIsMissing;
+            }
+         }
+         fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()] = entry;
+      }
+      return fLastResult[slot * RDFInternal::CacheLineStep<int>()];
+   }
+
+   ROOT::Detail::RDF::RColumnReaderBase *GetOrCreateColumnReader(TTreeReader *r, unsigned int slot)
+   {
+      // Try to check if there is an available reader from the column register first
+      // We do not check that the type of the reader matches the type of the input column of this node,
+      // because this node does not keep track of that anyway
+      if (auto *defineOrVariationReader = fColRegister.GetReaderUnchecked(slot, fColumnNames[0], fVariation))
+         return defineOrVariationReader;
+
+      // Check if we already inserted a reader for this column in the dataset column readers (RDataSource or Tree/TChain
+      // readers)
+      if (auto *datasetColReader = fLoopManager->GetDatasetColumnReader(slot, fColumnNames[0], typeid(void)))
+         return datasetColReader;
+
+      // Create a column reader that does not need type information. It is used only to check if an entry of the
+      // column can be read validly.
+      return fLoopManager->AddTreeColumnReader(
+         slot, fColumnNames[0], std::make_unique<ROOT::Internal::RDF::RTreeOpaqueColumnReader>(*r, fColumnNames[0]),
+         typeid(void));
+   }
+
+   void InitSlot(TTreeReader *r, unsigned int slot) final
+   {
+      fValues[slot] = GetOrCreateColumnReader(r, slot);
+      fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()] = -1;
+   }
+
+   // recursive chain of `Report`s
+   void Report(ROOT::RDF::RCutFlowReport &rep) const final { PartialReport(rep); }
+
+   void PartialReport(ROOT::RDF::RCutFlowReport &rep) const final
+   {
+      fPrevNodePtr->PartialReport(rep);
+      FillReport(rep);
```

Comment:
If `filtername` passed in the constructor is defaulted (ie. `""`) will there be any visible difference (besides the number of accepted vs rejected) between a `FilterMissing` and `FilterAvailable` ?

Reply:
The `Report` operation only works on named filters, i.e. a Filter node where the `filtername` is not empty. By not having a filter name argument in both `FilterMissiing` and `FilterAvailable`, I am practically preventing `Report` to even work with these two new functions. This is done on purpose for now, as I don't want to extend the functionality too much. I foresee this as an easy extension for after 6.34.00.